libr/bin/format/elf/elf.c

```diff
@@ -2962,7 +2962,8 @@ static int Elf_(fix_symbols)(ELFOBJ *bin, int nsym, int type, RBinElfSymbol **sy
 			while (!p->last) {
 				if (p->offset && d->offset == p->offset) {
 					p->in_shdr = true;
-					if (*p->name && strcmp (d->name, p->name)) {
+					if (*p->name && ((strlen(p->name) != strlen(d->name) && strncmp(d->name, p->name, strlen(p->name)) == 0) ||
+							(d->name && d->name[0] == '$'))) {
```

Comment:
I dont like this strlenncomparison, also internally this code is doing several more strlen calls which are totally unnecessary also as long as this conditional is becoming a bit complex move that into a static function with proper name.

Also always add a space before the (

Another issue i see is that you are checking for d->name not being null after dereferencing it

You can use r_str_startswith instead of that strncmo+strlen